### PR TITLE
Add views - Filter And Sort

### PR DESCRIPTION
**Pull request by jben94**
Hi,
Fix filter bug in adding views and sorting views.

### CODE REVIEW

File: www/api/class/centreon_home_customview.class.php [hunk 1]
```diff
@@ -69,9 +74,17 @@ public function getListSharedViews()
...
             'SELECT cvur2.custom_view_id FROM custom_view_user_relation cvur2 ' .
             'WHERE cvur2.user_id = ' . $centreon->user->user_id . ' ' .
-            'AND cvur2.is_consumed = 1) ';
+            'AND cvur2.is_consumed = 1) ' .
+            ($q != ''?'AND d.name like ? ':'') .
+            ') tmp ORDER BY tmp.name';
+
+        if (isset($this->arguments['q']) && $this->arguments['q'] != '') {
+            $stmt = $this->pearDB->prepare($query);
+            $dbResult = $this->pearDB->execute($stmt, array('%'.$q.'%'));
```
Comment (kduret) [inline www/api/class/centreon_home_customview.class.php]: Missing spaces between dots

File: www/api/class/centreon_home_customview.class.php [hunk 2]
```diff
@@ -69,9 +74,17 @@ public function getListSharedViews()
             'WHERE d.custom_view_id NOT IN (' .
             'SELECT cvur2.custom_view_id FROM custom_view_user_relation cvur2 ' .
             'WHERE cvur2.user_id = ' . $centreon->user->user_id . ' ' .
-            'AND cvur2.is_consumed = 1) ';
+            'AND cvur2.is_consumed = 1) ' .
+            ($q != ''?'AND d.name like ? ':'') .
```
Comment (kduret) [inline www/api/class/centreon_home_customview.class.php]: Missing spaces around ? and :

**Comment by jben94** [comment]
Hi @kduret 
I just changed my code.
Is it good for you?

**Comment by loiclau** [comment]
hi , thanks for your contribution

**Pull request merged**